cmd/protoc-gen-netlify-cms/main.go

```diff
@@ -412,8 +413,10 @@ func inferField(
 		}
 		return field, true
 	case protoField.Desc.Kind() == protoreflect.StringKind && !protoField.Desc.IsList():
-		field.Widget.WidgetType = &cmsv1.Widget_StringWidget{
-			StringWidget: &cmsv1.StringWidget{},
+		if field.Widget.WidgetType == nil {
```

Comment:
this looks like a side-effect of breaking the invariant of not proceeding on this code path if a widget has been annotated - if you want special casing for name fields, is there a way to do it without breaking this invariant?

Reply:
the code below only does special configuration for `name` field, so for normal string case it is fine?

or if you want, we can perhaps add an early return here if the field is not name and it has some widgetType already?

Reply:
made an update with another commit, this looks simpler and we also keep the old flow if the field is not a `name`